allensdk/brain_observatory/nwb/__init__.py

```diff
@@ -385,9 +433,23 @@ def add_stimulus_presentations(nwbfile, stimulus_table, tag='stimulus_epoch'):
             series.fillna('', inplace=True)
             stimulus_table[colname] = series.transform(str)
 
-    stimulus_table = setup_table_for_epochs(stimulus_table, ts, tag)
-    container = pynwb.epoch.TimeIntervals.from_dataframe(stimulus_table, 'epochs')
-    nwbfile.epochs = container
+    stimulus_names = stimulus_table['stimulus_name'].unique()
+    for stim_name in sorted(stimulus_names):
+        interval_description = (f"Presentation times and stimuli details "
+                                f"for '{stim_name}' stimuli")
+
+        presentation_interval = create_stimulus_presentation_time_interval(
+            name=f"{stim_name}_presentations",
+            description=interval_description,
+            columns_to_add=stimulus_table.columns
```

Comment:
Is the same set of columns added to each stimulus table? Some columns/properties seem to be relevant only to certain stimuli (e.g., orientation is relevant for gabors but not natural movies). This is fine to do but wastes space in each table.

Reply:
The whole point of requesting different TimeIntervals table was so that we could associate the appropriate parameters to each. It would be nice if we could condense this down so that a user can immediately understand what parameters are being manipulated for each stimulus type.

Reply:
@bendichter Alright, @rly had a good idea in our slack conversation, for how to condense down the columns, that I can try. I will be making that change in #1584.